lib/pwa.js

```diff
@@ -251,37 +251,63 @@ exports.findByEncodedStartUrl = function(encodedStartUrl) {
 };
 
 /**
- * Validates and saves the PWA
+ * Creates or Updates a PWA.
+ *
+ * Steps:
+ *  1) Validate Pwa
+ *  2) Update Pwa's Manifest
+ *  3) Save (to get the DB id for following steps)
+ *  4) Update PWA's MetadataDescription
+ *  5) Update PWA's Icon
+ *  6) Save
+ *  7) (in background) Update Lighthouse, Save & Send notifications for new PWAs.
  *
  * @param {Pwa} pwa to update
  * @return {Pwa} the updated PWA
  */
-exports.save = function(pwa) {
-  if (!pwa) {
-    return Promise.reject('pwa is null');
-  }
+exports.createOrUpdatePwa = function(pwa) {
+  const steps = [
+    _ => this.validatePwa(pwa),
```

Comment:
```
result => this.updatePwaManifest(result),
```
is a function with the following signature: `function(result) : Promise`. This is the same signature as this.updatePwaManifest has. Don't call the functions in steps, but simply add a reference to the function (note the lack of parentheses after the function name):

```
const steps = [
    this.validatePwa,
    this.updatePwaManifest,
    ...
  ];